File: 0342-power-of-four/0342-power-of-four.js
/**
 * @param {number} n
 * @return {boolean}
 */

function powFour(n ,x ) {

    if (n===x) return true;

    if(x>=n) return false;


    return powFour(n , 4 * x)
}
var isPowerOfFour = function(n) {

    if(n==1) return true



    return powFour(n , 4)
    
};

File: 0342-power-of-four/0342-power-of-four.py
class Solution:

    def isPower(self ,n , x):

        if n==x: return True

        if x>=n: return False


        return self.isPower(n , 4 * x)




    def isPowerOfFour(self, n: int) -> bool:

        if n==1: return True

        return self.isPower(n , 4)

        

        
        